### PR TITLE
Feat: get payments

### PR DESCRIPTION
**Pull request by laouali80**
### Description
Create an API endpoint to fetch all payments.
### Acceptance Criteria
- Implement the GET endpoint `api/v1/payments`.
- Ensure the verification of the user role.
- The endpoint should return a list of payments.
- Ensure proper error handling for invalid requests.
- Write unit tests to cover various cases including successful fetch, empty results, and error scenarios.
### Purpose
To provide an endpoint that allows fetching all payment records from the database. This will facilitate various operations such as displaying payment history to users, auditing payments, and generating reports.
### Requirements
The endpoint should follow RESTful principles.
Use appropriate HTTP status codes for responses.
Ensure the data returned is in JSON format.
Validate that the endpoint correctly fetches data from the database based on the given schema.
Ensure secure access to the endpoint  via authentication/authorization.	
### Expected Outcome
The endpoint `api/v1/payments` successfully returns a JSON array of payments with the following structure for each payment:
Response:
200 Record Found, returning all payments.
```json
{
  "Status Code": 200,
  "message": "Records Found",
  "payments":[
{
    "id": 1,
    "user_id": 123,
    "amount": 100.00,
    "currency": "Naira",
    "status": "completed",
    "method": "credit card",
    "transaction_id": "txn_123ABC",
    "created_at": "2023-01-01T12:00:00Z"
  },
  ...
] 
}
```

**401 Unauthorized if the request is not made by a superadmin.**
```json
{
  "Status Code": 41
  "message": "unauthorization"
}
```
### Unit Test
Test fetching all payments when there are multiple records.
Test fetching payments when the database is empty.
Test error handling.
Test the endpoint with proper authentication and authorization.

### CODE REVIEW

File: api/v1/routes/get_payments.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+def get_all_payments(db: Session = Depends(get_db), current_user: User = Depends(user_service.get_current_super_admin)):
+    
+    
+    payments = db.query(Payment).all()
+    
+    return success_response(
+        status_code=200,
+        message= 'Records Found',
+        data=payments
+    )
```
Comment (Goketech) [inline api/v1/routes/get_payments.py:24]: prefix payment in the url and fetch from base url

make query in services folder

**Review by Goketech — CHANGES_REQUESTED** [review]

File: api/v1/routes/get_payments.py (hunk 1 above)
Comment (laouali80) [reply api/v1/routes/get_payments.py]: Fix

File: alembic/versions/15d0a3738191_initial_migration.py
Comment (joboy-dev) [inline alembic/versions/15d0a3738191_initial_migration.py:1]: Why are you making changes to migrations please

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Check my comment and resolve merge conflicts

**Pull request closed without merging**